### DISCUSSION TITLE
Is there a way to generate hooks in a separated file using `near-operation-file` and `typescript-react-apollo`?

### DISCUSSION THREAD

**Discussion by bcarvalho89** [dotansimha/graphql-code-generator-community#499] [Q&A]
I'm trying to figure how generate two files using the `near-operation-file` and `typescrit-react-apollo`.
One file for the types and another only for hooks.

Is there a way to do this? The goal is to have for each `.gql` file a generated file for `types` and another for `hooks`.
This is my `generates` config:

```
    './src/': {
      documents: ['./src/**/*.gql.ts'],
      preset: 'near-operation-file',
      presetConfig: {
        extension: '.types.ts',
        baseTypesPath: 'types/globalApiTypes.ts',
        folder: '__generated__',
      },
      plugins: [
        {
          add: {
            content: `/** THIS FILE IS AUTO-GENERATED **/
    /** DO NOT EDIT **/
    /* eslint-disable */
                `,
          },
        },
        'typescript-operations',
        'typescript-react-apollo',
      ],
    },
```

**Comment by Urigo** [comment]
Sorry for not yet answering your direct question, but is there a reason you prefer to use that and not Codegen's recommended [client-preset](https://the-guild.dev/graphql/codegen/docs/guides/react-vue)
